vignettes/gsmExtensions.Rmd

```diff
@@ -0,0 +1,266 @@
+
+# Module Configuration
+
+`gsm` modules will typically be part of an R package and have a YAML configuration file - `{module_name}.yaml` - saved under `\inst\workflows`. Module config files include 3 key properties: 
+
+  -  `meta`: (required) Report metadata used for centralized library and fields referenced in `steps` section of workflow.
+    - This is not required in the `{report_name}.yaml` if the report is entirely comprised of `metric`s that have their own `yaml` configuration files. 
+  - `spec`: (required) Data table requirement for the report. Typically from `Mapped_*` or `Reporting_*` data, but occasionally from `Analytics_*` and `Reporting_*` data sources. 
+  - `steps`: (required) Functions to produce the reporting output from `Mapped_*` and/or `Reporting_*` data to final output (`html`, `csv` or `shinyApp`).
+   
+Detailed specifications for each of these sections are provided below and links to several example module configuration files are provided at the end of this vignette (TODO: Add this). 
+
+## Example Module Configuration YAMLs
+
+Here are links to several sample module configuration files:
+
+- [12 Standard gsm KRIs](https://github.com/Gilead-BioStats/gsm/tree/dev/inst/workflow/metrics) (e.g. [Adverse Event KRI Metric](https://github.com/Gilead-BioStats/gsm/tree/dev/inst/workflow/metrics/kri0001.yaml))
+- [Site-](https://github.com/Gilead-BioStats/gsm/blob/dev/inst/workflow/reports/report_kri_site.yaml) and [Country-level KRI Report](https://github.com/Gilead-BioStats/gsm/blob/dev/inst/workflow/reports/report_kri_country.yaml)
+- [GSM Deep Dive Shiny App]()
+
+## `meta` Specification
+
+The `meta` section of workflow YAML provides key metadata describing the module. It must include the following fields: 
+
+- `Name`: Name of the reporting output.
+- `Type`: "Report", "Metric" or "App"
+- `ModuleID`/`MetricID`: The unique ID for the module. `ModuleID` is used for reports and apps, while `MetricID` is used for metrics.
+- `Description`: A one-line description of the module specified in the workflow.
+- `Details`: A more detailed description of the module specified in the workflow.
+- `Repo`: Package repo and version. Should be compatible with the `repo` parameter in `devtools::install_github()`. 
```

Comment:
yes, i am hoping that there will be a process that auto-fills/updates this field- we will want to update it with each release since all yamls should be functional at time of release, so adding it to the checklist and/or making a `version_update()` function that updates the version in Description and in all yamls (and any other locations) would make the most sense. We could even trigger that function in a GHA to prepare for release. 

I was holding off making an issue for it since it's a bit in the weeds, but I'll add it now so that we have it. It's possible that there will be a gsm.template solution here, but i'm not sure at the moment.